core/src/test/java/org/apache/calcite/test/RelMetadataTest.java

```diff
@@ -2483,38 +2484,43 @@ private void checkPredicates(RelOptCluster cluster, RelOptTable empTable,
   }
 
   @Test void testExpressionLineageConjuntiveExpression() {
-    // empno is column 0 in catalog.sales.emp
-    // ename is column 1 in catalog.sales.emp
-    // deptno is column 7 in catalog.sales.emp
-    final RelNode rel = convertSql("select (empno = 1 or ename = 'abc') and deptno > 1 from emp");
+    final String sql = "select (empno = 1 or ename = 'abc') and deptno > 1 from emp";
+    final RelNode rel = convertSql(sql);
     final RelMetadataQuery mq = rel.getCluster().getMetadataQuery();
 
     final RexNode ref = RexInputRef.of(0, rel.getRowType().getFieldList());
-    final Set<RexNode> r = mq.getExpressionLineage(rel, ref);
-    assertThat(r.size(), is(1));
+    final Set<RexNode> r = Objects.requireNonNull(mq.getExpressionLineage(rel, ref));
```

Comment:
is `requireNonNull` needed here?

Reply:
The motivation is exactly the same: test failure should include meaningful information.
If you add `requireNonNull`, `getOnlyElement`, etc, etc, then you make it harder to understand the nature of the failure.

This case is trivial: the test expects a set with a single element. The assertion message is more or less OK.
Why adding more checks before the final `assertEquals`?
I believe, the final `assertEquals` should be more than enough to tell that "expected ... got null".